Tools/gittools/pre-commit.py

```diff
@@ -75,10 +80,10 @@ def run(self):
                 # don't check deleted files
                 continue
             (base, extension) = os.path.splitext(filepath)
-            if extension != ".py":
-                continue
-            if not self.check_file(filepath):
-                self.retcode = 1
+            if extension == ".py" and self.check_for_flake8_tag(filepath):
+                self.files_to_check_flake8.append(filepath)
+        if self.files_to_check_flake8:
+            self.check_flake8()
```

Comment:
Is there some reason to keep the state persistently in self?  i.e. do `self.retcode` and `files_to_check_flake8` really need to be in `self`?

i.e. this could just be
```
        return self.run_flake8(files_to_check)
```

Reply:
No reason. More of a future thinking if more checks were added. Do all of the checks and then return from pre-commit.